File: app/src/main/java/org/tat/sdksample/recommendedroute/model/RecRoutes.java
package org.tat.sdksample.recommendedroute.model;

import android.os.Parcel;
import android.os.Parcelable;
import org.th.tatsdk.route.list.TATGetRoutesResult;

import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.List;

public class RecRoutes implements Parcelable {

    List<RecRoute> recRoutes;

    public RecRoutes(TATGetRoutesResult[] result) {
            List<RecRoute> list = new ArrayList(result.length);
        for (TATGetRoutesResult item : result) {
            list.add(new RecRoute(
                    item.getId(),
                    item.getName(),
                    item.getInfo(),
                    item.getThumbnail(),
                    distanceToUnit(item.getDistance()),
                    item.getNumberOfDays()
            ));
        }
        this.recRoutes = list;
    }

    public List<RecRoute> getRecRoutes() {
        return recRoutes;
    }

    public static Creator<RecRoutes> getCREATOR() {
        return CREATOR;
    }

    protected RecRoutes(Parcel in) {
        recRoutes = in.createTypedArrayList(RecRoute.CREATOR);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(recRoutes);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<RecRoutes> CREATOR = new Creator<RecRoutes>() {
        @Override
        public RecRoutes createFromParcel(Parcel in) {
            return new RecRoutes(in);
        }

        @Override
        public RecRoutes[] newArray(int size) {
            return new RecRoutes[size];
        }
    };

    public static String distanceToUnit(double distance) {
        if (distance != 0.0) {
            if (distance >= 1000) {
                DecimalFormat decimalFormat = new DecimalFormat("#.#");
                return String.valueOf(decimalFormat.format(distance / 1000)).replace(".0", "") + "km.";
            } else {
                return String.valueOf(Integer.valueOf((int) distance)) + "m.";
            }
        } else {
            return "";
        }
    }
}

File: app/src/main/java/org/tat/sdksample/recommendedroute/model/RecRoutes.kt
package org.tat.sdksample.recommendedroute.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import org.th.tatsdk.route.TATFindRoutesResult
import java.text.DecimalFormat

@Parcelize
data class RecRoutes(var recRoutes: List<RecRoute>? = null) : Parcelable {

    fun transform (result: Array<TATFindRoutesResult>) : RecRoutes {
        val list = ArrayList<RecRoute>(result.size)
        for (item in result) {
            list.add(RecRoute(
                    item.id,
                    item.name,
                    item.introduction,
                    item.thumbnailUrl,
                    distanceToUnit(item.distance),
                    item.numberOfDays?:0
            ))
        }
        return RecRoutes(list)
    }

    fun distanceToUnit(distance: Double): String {
        if (distance != 0.0) {
            if (distance >= 1000) {
                val decimalFormat = DecimalFormat("#.#")
                return decimalFormat.format(distance / 1000).toString().replace(".0", "") + "km."
            } else {
                return Integer.valueOf(distance.toInt()).toString() + "m."
            }
        } else {
            return ""
        }
    }
}



